src/lapack/Makefile

```diff
@@ -90,22 +90,89 @@ OBJS = ${LAPACK_FILES:.c=.o}
 # Targets
 #################################################################
 
-all: ${OBJS} dlamch.o
+all: lapack_OBJS dlamch_obj
 
 install: all
 	cp -fR $(srcdir)/_hypre_lapack.h $(HYPRE_INC_INSTALL)
 
-clean:
+clean: clean_mup
 	rm -rf *.o
 
 distclean: clean
 
-#################################################################
-# Rules
-#################################################################
+ifeq (${MP_BUILD}, 1) 
+clean_mup:
+	rm -rf ${MuP_OBJDIR}	
+else
+clean_mup:
+endif
+
+##################################################################
+# Build Rules
+##################################################################
+
+ifeq (${MP_BUILD}, 1) 
+
+MuP_OBJDIR = MuP_obj.dir
+MuP_OBJDIR_single = ${MuP_OBJDIR}/${join ${shell basename ${CURDIR}},_flt}
+MuP_OBJDIR_double = ${MuP_OBJDIR}/${join ${shell basename ${CURDIR}},_dbl}
+MuP_OBJDIR_longdouble = ${MuP_OBJDIR}/${join ${shell basename ${CURDIR}},_ldbl}
+#
+#MKDIR_P = mkdir -p
+#mkmpdir: 
+#	${MKDIR_P} ${MuP_OBJDIR}
+#
+ADD_OBJS_single = ${addprefix ${MuP_OBJDIR_single}/, ${notdir ${OBJS}}}
```

Comment:
@falgout, I agree we could move most of this into Makefile.config, as the approach is similar for many of the subdirectories. I also think it can be simplified or cleaned-up a bit. In addition, when we include the multiprecision methods, they will have to be built differently (only once) and so like the utilities directory, the resulting makefile will have slightly more diffs. Take a look at the seq_mv subdirectory on the [mp-buil-all](https://github.com/hypre-space/hypre/blob/mp-build-all/src/seq_mv/Makefile) branch to see such an example. If you are interested in taking a pass on incorporating it within the Makefile.config file, let me know. Otherwise I can take a pass at it and iterate with you (and everyone else).